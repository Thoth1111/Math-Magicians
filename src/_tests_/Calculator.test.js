import * as React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

import Calculator from '../components/calculator';
import CalcButton from '../components/calc-button';

describe('Calculator display', () => {
    it('renders the go-time text', () => {
        render(<Calculator />);
        expect(screen.getByText(/It's Math Time!!/)).toBeInTheDocument();
    })
    it('renders all the calculator UI buttons', () => {
        render(<Calculator />);
        expect(screen.getAllByRole('button')).toHaveLength(19);
    });
    it('renders the complete UI correctly', () => {
        const tree =  renderer.create(<Calculator />).toJSON();
        expect(tree).toMatchSnapshot();
    });    
});

describe('Calculator interaction', () => {
    it('calls the handleButtonClick function', () => {
        const handleButtonClick = jest.fn();
        render(<CalcButton value="AC" onClick={handleButtonClick} />);
        userEvent.click(screen.getByText('AC'));
        expect(handleButtonClick).toHaveBeenCalled();
    });
    it('calls the handleButtonClick function', () => {
        const handleButtonClick = jest.fn();
        render(<CalcButton value="7" onClick={handleButtonClick} />);
        userEvent.click(screen.getByText('7'));
        expect(handleButtonClick).toHaveBeenCalled();
    });
    it('calls the handleButtonClick function', () => {
        const handleButtonClick = jest.fn();
        render(<CalcButton value="4" onClick={handleButtonClick} />);
        userEvent.click(screen.getByText('4'));
        expect(handleButtonClick).toHaveBeenCalled();
    });
    it('calls the handleEqualsClick function', () => {
        const handleEqualsClick = jest.fn();
        render(<CalcButton value="=" onClick={() => handleEqualsClick()} />);
        userEvent.click(screen.getByText('='));
        expect(handleEqualsClick).toHaveBeenCalled();
    });

})


