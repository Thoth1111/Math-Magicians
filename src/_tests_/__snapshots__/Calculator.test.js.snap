// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator display renders the complete UI correctly 1`] = `
Array [
  <h2
    className="go-time-text"
  >
    It's Math Time!!
  </h2>,
  <section
    className="calc-UI"
  >
    <div
      className="display"
    >
      0
    </div>
    <div
      className="keys"
    >
      <button
        className="calc-Btns"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="calc-Btns"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="calc-Btns"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="calc-Btns"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="keys"
    >
      <button
        className="calc-Btns"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="calc-Btns"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="calc-Btns"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="calc-Btns"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="keys"
    >
      <button
        className="calc-Btns"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="calc-Btns"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="calc-Btns"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="calc-Btns"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="keys"
    >
      <button
        className="calc-Btns"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="calc-Btns"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="calc-Btns"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="calc-Btns"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="keys"
    >
      <button
        className="calc-Btns"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="calc-Btns"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="calc-Btns"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </section>,
]
`;
